test/over_react/component/shared_error_boundary_tests.dart

```diff
@@ -226,17 +252,22 @@ void sharedErrorBoundaryTests(BuilderOnlyUiFactory builder) {
     void _setCallbackVarValues() {
       expect(calls, isNotNull, reason: 'test setup sanity check');
       expect(calls, isNotEmpty, reason: 'test setup sanity check');
-      expect(calls[0].keys.single, 'onComponentDidCatch', reason: 'test setup sanity check');
-      expect(jacket.getDartInstance().state.hasError, isTrue, reason: 'test setup sanity check');
+
+      if (isWrapper && calls.length > 3) {
+        // Only the last 2 calls to `onComponentDidCatch` and `onComponentIsUnrecoverable`
+        // belong to the wrapping error boundary.
+        calls = calls.sublist(calls.length - 2);
+      }
 
       final componentDidCatchCallbackArguments = calls[0]['onComponentDidCatch'];
       if (componentDidCatchCallbackArguments != null) {
         errorSentToComponentDidCatchCallback = componentDidCatchCallbackArguments[0];
         errorInfoSentToComponentDidCatchCallback = componentDidCatchCallbackArguments[1];
       }
 
+      // At this point the component is "Unrecoverable"
       if (calls.length > 1) {
-        expect(calls[1].keys.single, 'onComponentIsUnrecoverable', reason: 'test setup sanity check');
+        expect(calls[calls.length - 1].keys.single, 'onComponentIsUnrecoverable', reason: 'test setup sanity check');
```

Comment:
#nit could use `.last`
```suggestion
        expect(calls.last.keys.single, 'onComponentIsUnrecoverable', reason: 'test setup sanity check');
```

Reply:
well la-di-da mr dart is awesome... back in my day when javascript ruled the world we didnt have the luxury of your fancy smanchy `.last` you had to put in the effort and really think, i need the last item in this array so its length... but its indexed from 0 so i need to subtract 1... thats real dev...